### PR TITLE
Opensearch template index

### PR DESCRIPTION
**Pull request by AmartC**
This PR adds logic to the Opensearch Updating Service by creating the templates index which stores the template, the template cluster ID as well as the most frequently occurring log message associated with that template. It receives this information through the Opni DRAIN service.

### CODE REVIEW

File: aiops/opni-opensearch-update-service/opensearch-update-service/app/main.py [hunk 1]
```diff
@@ -122,6 +167,19 @@ async def update_logs(es, df):
...
+        exists = await es.indices.exists("templates")
+        if not exists:
+            await es.indices.create(index="templates")
+
+    except TransportError as exception:
+        logging.info(f"Error in es indices {exception}")
+        if exception.status_code == "N/A":
+            logging.info("Elasticsearch connection error")
+
+
```
Comment (dbason) [inline aiops/opni-opensearch-update-service/opensearch-update-service/app/main.py:182]: This should be managed as part of the operator code we have for creating indices.  We should include an index template if the document in the index will be a standard format that we control.

As an example here is where we create some index templates - https://github.com/rancher/opni/blob/main/pkg/resources/opnicluster/elastic/indices/indices.go#L193-L219

And here is where we create an index:
https://github.com/rancher/opni/blob/main/pkg/resources/opnicluster/elastic/indices/indices.go#L262-L266

**Review by dbason — CHANGES_REQUESTED** [review]

File: aiops/opni-opensearch-update-service/opensearch-update-service/app/main.py [hunk 2]
```diff
@@ -23,10 +23,11 @@
     "ctx._source.anomaly_predicted_count += 1; ctx._source.opnilog_anomaly = true;"
 )
 
-async def doc_generator(df):
+async def doc_generator(df, index, op_type="index"):
     main_doc_keywords = {"_op_type", "_index", "_id", "doc"}
-    df["_op_type"] = "update"
-    df["_index"] = "logs"
+    if op_type != "index":
```
Comment (tybalex) [inline aiops/opni-opensearch-update-service/opensearch-update-service/app/main.py:28]: unnecessary `if`

File: aiops/opni-opensearch-update-service/opensearch-update-service/app/main.py [hunk 3]
```diff
@@ -23,10 +23,11 @@
     "ctx._source.anomaly_predicted_count += 1; ctx._source.opnilog_anomaly = true;"
 )
 
-async def doc_generator(df):
+async def doc_generator(df, index, op_type="index"):
```
Comment (tybalex) [inline aiops/opni-opensearch-update-service/opensearch-update-service/app/main.py:26]: default op_type should be `update`.

**Review by tybalex — CHANGES_REQUESTED** [review]
pls clean up debugging loggers.

File: aiops/opni-opensearch-update-service/opensearch-update-service/app/main.py (hunk 3 above)
Comment (AmartC) [reply aiops/opni-opensearch-update-service/opensearch-update-service/app/main.py:26]: By default, if _op_type isn't specified, it's index.

**Review by dbason — APPROVED** [review]

**Review by dbason — APPROVED** [review]

**Review by tybalex — APPROVED** [review]

**Review by tybalex — APPROVED** [review]

**Pull request merged**